app/jobs/issue_event_job.rb

```diff
@@ -2,62 +2,83 @@ class IssueEventJob < ActiveJob::Base
 
   around_perform :guard_against_double_events
 
+  class_attribute :_action
+  class_attribute :_cache_key
+  class_attribute :_timestamp
+  class_attribute :_identifier
+
   def self.action(action)
-    @_action = action
+    self._action = action
   end
 
-  @_timestamp = Proc.new { Time.now.utc.iso8601 }
-  def self.timestamp(override=nil)
-    if override
-      @_timestamp = override
-    else
-      @_timestamp
-    end
+  self._timestamp = Proc.new { Time.now.utc.iso8601 }
+
+  def self.timestamp(override)
+    self._timestamp = override
   end
+
+  self._identifier = ->(p){ p['issue']['number'] }
+
   def self.identifier(override=nil)
-    @_identifier = override
+    self._identifier = override
+  end
+
+
+  self._cache_key = ->(message) { "#{message[:meta][:action]}.#{message[:meta][:user]["login"]}.#{message[:meta][:identifier]}.#{message[:meta][:timestamp]}" }
```

Comment:
Slightly different, but its hard to see. class method so these are "named" more apparently